src/Nethermind/Nethermind.Core/KeyValueStoreExtensions.cs

```diff
@@ -108,7 +108,7 @@ public static void Set(this IWriteOnlyKeyValueStore db, Hash256 key, in CappedAr
         {
             if (value.IsUncapped && db.PreferWriteByArray)
             {
-                db.Set(key.Bytes, value.ToArray(), writeFlags);
+                db.PutSpan(key.Bytes, value.AsSpan(), writeFlags);
```

Comment:
Nope, see the `PreferWriteByArray` there. Well, I guess seem Caching store is removed, this should not be needed anymore.